options.c

```diff
@@ -1369,6 +1369,10 @@ int parse_arguments(int *argc_p, const char ***argv_p)
 	/* TODO: Call poptReadDefaultConfig; handle errors. */
 
 	pc = poptGetContext(RSYNC_NAME, argc, argv, long_options, 0);
+	if (pc == NULL) {
```

Comment:
while correct, checking for NULL is also a layering violation, as the object returned is opaque.

doing instead `if (!pc)` as done in the `cleanup` label should be better IMHO, but adding something to the documentation of [poptGetContext()](/rpm-software-management/popt) that validates that use case even better.